pkg/ddl/partition.go

```diff
@@ -2186,8 +2186,8 @@ func (w *worker) rollbackLikeDropPartition(jobCtx *jobContext, job *model.Job) (
 	var dropIndices []*model.IndexInfo
 	for _, indexInfo := range tblInfo.Indices {
 		if indexInfo.Unique &&
```

Comment:
What's the special for unique index here?

Reply:
Currently the Global Index can only be unique, so is just filters out indexes that are not affected.

Reply:
This will affect non-global unique index or not?

Reply:
So this path also handles `ALTER TABLE t PARTITION BY ... [UPDATE INDEXES (...)]` so yes, it also handles non-global unique indexes that might be changed from/to global indexes.

Reply:
Will it handle the case of `unique index` -> `non global unique index`? Maybe it's no need to process.

Reply:
After I debuged, maybe `indexInfo.State == model.StateWriteOnly` is enough. Am I right?

Reply:
Yes that should be enough, or even use the new DDLChangedIndex map. It should be safe for ADD PARTITION as well, since it also using this function.